### ISSUE TITLE
Add a ShowSubcommandHelpAndExit()

### ISSUE THREAD

**Issue by bodgit** [urfave/cli#1112]
## Checklist

* [x] Are you running the latest v2 release? The list of releases is [here](https://github.com/urfave/cli/releases).
* [x] Did you check the manual for your release? The v2 manual is [here](https://github.com/urfave/cli/blob/master/docs/v2/manual.md)
* [x] Did you perform a search about this feature? Here's the [Github guide](https://help.github.com/en/github/managing-your-work-on-github/using-search-to-filter-issues-and-pull-requests) about searching.

## What problem does this solve?

There is a `ShowAppHelpAndExit()` and a `ShowCommandHelpAndExit()` but there's currently no `ShowSubcommandHelpAndExit()`.

## Solution description

I usually use at the top of my command actions something like this:
```go
if c.NArg() < 2 {
        cli.ShowCommandHelpAndExit(c, c.Command.FullName(), 1)
}
```
So if someone doesn't supply the correct number of positional arguments they get the help and it exits with a non-zero code.

I tried to use sub-commands for the first time and found I can't do this.

## Describe alternatives you've considered

I've tried implementing as:
```go
if c.NArg() < 2 {
        cli.ShowSubcommandHelp(c)
        return cli.NewExitError(nil, 1)
}
```
but that doesn't work very well.

**Comment by lynncyrin** [comment]
@bodgit what about your alternative doesn't work very well?

**Comment by meatballhat** [comment]
Actually closed via #1183
